text/0000-engines-component-dependencies.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes to allow the declaration of components as dependencies of engines.
+
+# Motivation
+
+Currently, the only way to allow engines to share components with their consuming application/engine is to extract the component and package it as a standalone add-on. The add-on can then be installed by any application/engine that requires it.
+
+While this approach is a perfectly viable option in some circumstances, the viability begins to breakdown in distributed applications where the developer may not have control of some or all of the engines being consumed.
+
+In such circumstances it becomes difficult to ensure consistency throughout the application as each engine could theoretically be using a different version of the same component, potentially resulting in different areas of the application having different UI/UX for the same component/functionality.
+
+Allowing engines to declare components as dependencies would resolve this issue of consistency. If any changes were made to components in the main application, these changes would be able flow down through to each of the mounted engines without requiring the engine developers to change any of their code.
+
+# Detailed design
```

Comment:
I think I'd need a bit of assistance here with regards to outputted code etc, I don't have a great deal of knowledge of the internals of Ember itself or the code that is outputted from EmberCLI.

With regards to the context for a component, I can't think of a scenario where the context would be anything other than the engine itself.